improve documentation (#58)
CodeIntelligenceTesting/cifuzz: docs/Glossary.md

# Glossary
To make sure we are talking about the same things we rely on the following definitions:

## Fuzzer aka Fuzzing Engine
Is a piece of software that generates input to feed into the system under test (SUT) via the fuzz targets.
A coverage guided fuzzer will also collect information about the covered code during the execution and will try to 
improve coverage by adjusting the inputs accordingly.

## Fuzz Target
Usally a function that takes an array of bytes from the fuzzer and calls functions from the system under test.

## Fuzzing aka Fuzzing Run

> Fuzzing describes the act of running the fuzzer against a fuzz target.

Reviewer: We should get rid of the "aka"s here IMO. We'll have to have a discussion about which terms to keep and also about "Fuzz Target" vs "Fuzz Test".

Author: I removed it and I agree to the Test vs Target topic... we should discuss it and adjust the glossary then

## Corpus
A set of useful test inputs

### Seed Corpus

> A small corpus usually provided by the user. Findings can be added to this corpus to use it during regression tests.

Reviewer: ```suggestion
A small corpus usually provided by the user. Findings can be added to this corpus for use in regression tests.
```

Author: done

### Generated Corpus

> While executing a fuzzing run the fuzzer might discover useful inputs. These inputs are added to the generated corpus.

Reviewer: ```suggestion
The generated corpus contains interesting inputs found by the fuzzer while running the fuzz test.
```

Author: done

## Replayer

> A Replayer just applies inputs stored in the (seed) corpus to the fuzz target without generating new input.

Reviewer: ```suggestion
A replayer runs inputs from the seed corpus against the fuzz target without generating new input.
```
(using wording similar to https://go.dev/doc/fuzz/#command-line-output)

> It can be used during debugging and/or for regression testing.

Reviewer: ```suggestion
It can be used for debugging and/or regression testing.
```

Author: done

## Sanitizer

> Santizers use code instrumentation to detect bugs during the execute of an application. 

Author: done

## Finding
A finding describes a bug or vulnerability found by the fuzzer and includes the input that causes it.

## Metrics
While using **cifuzz** you will get in touch with metrics descriping the progress or outcome of a fuzzing run. 
Some of them might have different meanings depending on the context, previous knowledge or might even be used 

> differntly by the various fuzzing engines. To avoid confusion we provide the following definitions:

Reviewer: ```suggestion
differently by the various fuzzing engines. To avoid confusion we provide the following definitions:
```

Author: done
